prowler/providers/aws/services/ec2/ec2_securitygroup_allow_ingress_from_internet_to_any_port/ec2_securitygroup_allow_ingress_from_internet_to_any_port.py

```diff
@@ -35,8 +35,26 @@ def execute(self):
                         if check_security_group(
                             ingress_rule, "-1", ports=None, any_address=True
                         ):
-                            report.status = "FAIL"
-                            report.status_extended = f"Security group {security_group.name} ({security_group.id}) has at least one port open to the Internet."
+                            report.status_extended = f"Security group {security_group.name} ({security_group.id}) has at least one port open to the Internet but is exclusively attached to none or to an allowed network interface type."
+                            for eni in security_group.network_interfaces:
+                                if eni.type not in ec2_client.audit_config.get(
+                                    "ec2_allowed_interface_types",
+                                    [],
+                                ) or (
+                                    hasattr(eni, "attachment")
+                                    and isinstance(eni.attachment, dict)
+                                    and "InstanceOwnerId" in eni.attachment
+                                    and eni.attachment["InstanceOwnerId"]
+                                    not in ec2_client.audit_config.get(
+                                        "ec2_allowed_instance_owners",
+                                        [],
+                                    )
+                                ):
+                                    report.status = "FAIL"
+                                    report.status_extended = f"Security group {security_group.name} ({security_group.id}) has at least one port open to the Internet and is not exclusively attached to an allowed network interface type."
```

Comment:
What about the `InstanceOwnerId`? Maybe we should put something into the `report.status_extended?

What do you think? @kagahd @sergargar

Reply:
I think that the logic is still wrong even though the unit tests pass. Let me explain later but don't merge yet.

Reply:
For sure, I'll add the `no-merge` label.

Reply:
I cleaned up the code to make the logic easier to understand even though it stays the same boolean logic: if `eni_type` is allowed **OR** `eni_owner` is allowed (or both), then the check PASSes even though the Security Group is open to the Internet for a given port.
I added also still more unit tests to cover all cases.
I think it's now ready to be merged.

Reply:
> For sure, I'll add the `no-merge` label.

@jfagoagas, the `no-merge` label can be removed from my point of view. Thanks!